### Failed to build on `loongson3`

#### Error details

Environment: loongson3 buildbot, new BuildKit instance with Ciel
Errpoint: linking failed for `tigervnc`
Additional notes: There's 2 additional packages should built before building this package. (`tigervnc`, `xorg-server`)
Output:

```
[ 98%] Building CXX object tests/unit/CMakeFiles/conv.dir/conv.cxx.o
[100%] Linking CXX executable conv
[100%] Built target hostport
[100%] Built target conv
[100%] Built target x0vncserver
/bin/ld: /tmp/vncviewer.GYfECk.ltrans1.ltrans.o: .symtab local symbol at index 350 (>= sh_info of 45)
/bin/ld: /tmp/vncviewer.GYfECk.ltrans1.ltrans.o: error adding symbols: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [vncviewer/CMakeFiles/vncviewer.dir/build.make:319: vncviewer/vncviewer] Error 1
make[1]: *** [CMakeFiles/Makefile2:909: vncviewer/CMakeFiles/vncviewer.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] Built target encperf
make: *** [Makefile:149: all] Error 2
[ERROR]: Build error detected ^o^
In file included from /bin/autobuild:64
                 from /usr/lib/autobuild3/lib/base.sh:118
                 from /usr/lib/autobuild3/lib/base.sh:122
                 from /usr/lib/autobuild3/proc/50-build_exec.sh:27
                 from /usr/lib/autobuild3/build/00-self.sh:12
                 from /usr/lib/autobuild3/lib/arch.sh:75
                 from /usr/lib/autobuild3/lib/base.sh:122
autobuild/build:4: In function `.':
autobuild/build:4: error: command exited with 2
>   5 | make
    6 | make install DESTDIR="$PKGDIR"
```

Topic is a non-merge-able state as it involves off-topic packages.

--- a/extra-network/tigervnc/autobuild/defines
+++ b/extra-network/tigervnc/autobuild/defines
@@ -3,3 +3,6 @@ PKGSEC=net
 PKGDEP="fltk gnutls libgcrypt mesa libjpeg-turbo pixman x11-app xkeyboard-config"
 BUILDDEP="x11-font x11-app x11-proto cmake imagemagick nasm"
 PKGDES="Suite of VNC servers and clients"
+
+NOLTO__LOONGSON3=1
+

--- a/extra-network/xrdp/autobuild/patches/0003-add-arch-for-ppc64.patch
+++ b/extra-network/xrdp/autobuild/patches/0003-add-arch-for-ppc64.patch
@@ -1,0 +1,16 @@
+diff --git a/common/arch.h b/common/arch.h
+index ccccfa5a..6bf1d9dd 100644
+--- a/common/arch.h
++++ b/common/arch.h
+@@ -80,7 +80,9 @@ typedef int bool_t;
+     defined(__AIX__) || defined(__mips__) || \
+     defined(__ia64__) || defined(__arm__) || \
+     (defined(__PPC__) && defined(__BIG_ENDIAN__)) || \
+-    (defined(__ppc__) && defined(__BIG_ENDIAN__))
++    (defined(__ppc__) && defined(__BIG_ENDIAN__)) || \
++    (defined(__PPC64__) && defined(__LITTLE_ENDIAN__)) || \
++    (defined(__ppc64__) && defined(__LITTLE_ENDIAN__))
+ #define NEED_ALIGN
+ #elif defined(__x86__) || defined(__x86_64__) || \
+       defined(__AMD64__) || defined(_M_IX86) || defined (_M_AMD64) || \
+

